src/subscription_manager/gui/registergui.py

```diff
@@ -1992,6 +2114,24 @@ def _refresh(self, callback):
         except Exception:
             self.queue.put((callback, None, sys.exc_info()))
 
+    def __unregister_consumer(self, consumer_uuid, server_info):
```

Comment:
Instead of passing a server_info in, could we pass a already constructed UEP object? (One using the
last_server_info). That would avoid having set/reset cp_provider.

Not sure if a UEP has too much state to be passing to a thread (or more accurately, if it matters).